.github/workflows/ci.yml

```diff
@@ -194,24 +195,53 @@ jobs:
       - uses: actions/checkout@v4
         with: { ref: "${{ env.GIT_COMMIT }}" }
       - uses: ./.github/ci-setup-action
+      - name: Determine runner type
+        id: runner_type
+        run: |
+          runner=$(./scripts/select_runner.sh ${{ matrix.test }})
+          echo "type=$runner" >> $GITHUB_OUTPUT
       - name: Setup and Test
         timeout-minutes: 40
         uses: ./.github/ensure-tester-with-images
         env:
           USERNAME: ${{ needs.configure.outputs.username }}
         with:
-          runner_type: ${{ contains(matrix.test, 'prover') && '64core-tester-x86' || '8core-tester-x86' }}
+          runner_type: ${{ steps.runner_type.outputs.type }}
           builder_type: builder-x86
-          # these are copied to the tester and expected by the earthly command below
-          # if they fail to copy, it will try to build them on the tester and fail
           builder_images_to_copy: aztecprotocol/aztec:${{ env.GIT_COMMIT }} aztecprotocol/end-to-end:${{ env.GIT_COMMIT }}
-          # command to produce the images in case they don't exist
           builder_command: scripts/earthly-ci ./yarn-project+export-e2e-test-images
           run: |
             set -eux
             cd ./yarn-project/end-to-end/
             export FORCE_COLOR=1
-            ../../scripts/earthly-ci -P --no-output +${{ matrix.test }}
+
+            # Handle env vars for certain tests
+            if [[ "$matrix.test" == *"fake_proofs"* ]]; then
+              export FAKE_PROOFS=1
+            fi
+            if [[ "$matrix.test" == *"prover_full"* ]]; then
+              export HARDWARE_CONCURRENCY=32
+            fi
+
```

Comment:
I was thinking that we wouldn't need conditions here but instead have the commands listed in the test definition txt file

Reply:
ah, as in move this conditional logic inside `e2e_test.sh`?

Reply:
or just extend e2e_test_list to an object format where we define specific env vars & start command ig

Reply:
Ideally the script would let people run these locally as well, but a txt format works